moveit_core/collision_detection/include/moveit/collision_detection/collision_env.h

```diff
@@ -47,7 +47,10 @@ namespace collision_detection
 {
 MOVEIT_CLASS_FORWARD(CollisionEnv);  // Defines CollisionEnvPtr, ConstPtr, WeakPtr... etc
 
-/** \brief Provides the interface to the individual collision checking libraries. */
+/** \brief Provides the interface to the individual collision checking libraries.
+ * Some of these functions are impure, they depend on some shared variable .
+ *  The world_ variable for example is controlling the return value though it
```

Comment:
what return value? world_ can be set using the class API

Reply:
I was handling the concept with an academic [approach](https://www.syncfusion.com/blogs/post/pure-and-impure-functions-in-javascript-a-complete-guide.aspx#:~:text=An%20impure%20function%20is%20a,output%20for%20the%20same%20input.). A pure function is a function that returns the same value for the same input. In this class, the returned value "however value" in some functions depends on the state of the world_ variable.
The pull request as named was to just detect those functions, not enhance the code or put solutions to this at the current time. 
However, thank you for your comments it's really helpful!

Reply:
Depending on a class member doesn't make them impure. You don't seem to understand the term. The link points to some blog post, no academic approach visible. I'm closing this